src/main/java/org/ow2/proactive/catalog/dto/CatalogObjectMetadata.java

```diff
@@ -143,4 +151,114 @@ private String getProjectNameIfExistsOrEmptyString() {
         return projectNameIfExists.map(Metadata::getValue).orElse("");
     }
 
+    private LinkedHashMap<String, LinkedHashMap<String, JobVariable>> setVariablesOrder(List<Metadata> metadataList) {
+        LinkedHashMap<String, LinkedHashMap<String, JobVariable>> variablesOrder = new LinkedHashMap<>();
+
+        LinkedHashSet<String> groups = new LinkedHashSet<>();
+
+        // This item corresponds to the variables that have no groups - They must be added first
+        groups.add("");
+
+        // Get all variables with groups
+        groups.addAll(metadataList.stream()
+                                  .filter(metadata -> metadata.getLabel().equals("variable_group") &&
+                                                      !metadata.getValue().equals("") && metadata.getValue() != null)
+                                  .map(Metadata::getValue)
+                                  .collect(Collectors.toCollection(LinkedHashSet::new)));
+
```

Comment:
you should use the following map to improve the search:
`Map<String, String> variablesToGroups = metadataList.stream().filter(metadata -> metadata.getLabel().equals("variable_group") &&
                                                      !metadata.getValue().equals("") && metadata.getValue() != null).collect(Collectors.toMap(metadata -> metadata.getKey(), metadata -> metadata.getValue()));`

using this, checkIfTheVariableHasNoGroup becomes `!variablesToGroups.containsKey(variableName)`

Reply:
you can also use this map to find all variables belonging to a group